pkg/reconciler/buildrun/resources/conditions.go

```diff
@@ -109,15 +109,27 @@ func UpdateBuildRunUsingTaskRunCondition(ctx context.Context, client client.Clie
 					buildRun.Status.FailureDetails.Location.Container = failedContainer.Name
 				}
 			} else if failedContainer != nil {
+
 				//nolint:staticcheck // SA1019 we want to give users some time to adopt to failureDetails
 				buildRun.Status.FailureDetails.Location.Container = failedContainer.Name
-				message = fmt.Sprintf("buildrun step %s failed in pod %s, for detailed information: kubectl --namespace %s logs %s --container=%s",
-					failedContainer.Name,
-					pod.Name,
-					pod.Namespace,
-					pod.Name,
-					failedContainer.Name,
-				)
+
+				if failedContainer.Name == "step-image-processing" && exitCode == 22 {
+					reason = buildv1beta1.BuildRunStateVulnerabilitiesFound
+					message = fmt.Sprintf("Vulnerabilities have been found in the image which can be seen in the buildrun status. For detailed information,see kubectl --namespace %s logs %s --container=%s",
```

Comment:
Will there be reasonable output in the container logs?

Reply:
Not really, just the vulnerabilities list. and also the error, but in that case exit code won't be 22.